core/trino-main/src/main/java/io/trino/execution/QueryManagerConfig.java

```diff
@@ -679,16 +680,16 @@ public QueryManagerConfig setMaxTasksWaitingForExecutionPerQuery(int maxTasksWai
     }
 
     @Min(1)
-    public int getMaxTasksWaitingForNodePerStage()
+    public int getMaxTasksWaitingForNodePerQuery()
     {
-        return maxTasksWaitingForNodePerStage;
+        return maxTasksWaitingForNodePerQuery;
     }
 
-    @Config("max-tasks-waiting-for-node-per-stage")
-    @ConfigDescription("Maximum possible number of tasks waiting for node allocation per stage before scheduling of new tasks for stage is paused")
-    public QueryManagerConfig setMaxTasksWaitingForNodePerStage(int maxTasksWaitingForNodePerStage)
```

Comment:
Do we need to add `@LegacyConfig` for existing user? or User will have to change the configuration?

Reply:
I could add that - not sure if should given property name is just plain wrong - but maybe the way to go still.